pip_audit/_dependency_source/requirement.py

```diff
@@ -97,20 +98,26 @@ def collect(self) -> Iterator[Dependency]:
             # In order to get around this, we're going to copy each input into a
             # a corresponding temporary file and then pass that set of files
             # into `pip`.
-            tmp_files = []
-
-            # For each input file, copy it to one of our temporary files.
-            # Ensure we flush so our writes are visible to `pip`.
             for filename in self._filenames:
-                tmp_file = stack.enter_context(NamedTemporaryFile(mode="w"))
+                # Deliberately pass `delete=False` so that our temporary file doesn't get
+                # automatically deleted on close. We need to close it so that `pip` can
+                # use it however, we obviously want it to persist.
+                tmp_file = NamedTemporaryFile(mode="w", delete=False)
                 with filename.open("r") as f:
                     shutil.copyfileobj(f, tmp_file)
-                    tmp_file.flush()
-                tmp_files.append(tmp_file)
+
+                # Close the file since it's going to get re-opened by `pip`
+                tmp_file.close()
+                tmp_files.append(tmp_file.name)
 
             # Now pass the list of temporary filenames into the rest of our
             # logic.
-            yield from self._collect_from_files([Path(f.name) for f in tmp_files])
+            yield from self._collect_from_files([Path(f) for f in tmp_files])
+        finally:
+            # Since we disabled automatically deletion for these temporary files, we need to
+            # manually delete them on the way out.
+            for t in tmp_files:
+                os.unlink(t)
```

Comment:
This _shouldn't_ be necessary: with `delete=False` the `NamedTemporaryFile` won't unlink on `close()`, but it should still unlink when the context that manages it is destroyed. I'll test and confirm.

Reply:
Nope, I was wrong here -- `delete=False` means that `NamedTemporaryFile` doesn't delete at all.